### PR TITLE
Make suppresed exceptions loggable through logback

### PR DESCRIPTION
**Pull request by fsamuel-bs**
See https://jira.qos.ch/browse/LOGBACK-1027. Logback doesn't work well with circular dependencies.

**Goals (and why)**: Make logback work with `SuppressedException`.

**Implementation Description (bullets)**: -

**Concerns (what feedback would you like?)**: -

**Where should we start reviewing?**: -

**Priority (whenever / two weeks / yesterday)**: -

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2242)

### CODE REVIEW

**Comment by fsamuel-bs** [comment]
Should fix #2236.

File: atlasdb-client/src/main/java/com/palantir/atlasdb/util/SuppressedException.java [hunk 1]
```diff
@@ -19,14 +19,14 @@
...
-        super(message, cause);
+    SuppressedException(String message) {
+        super(message);
     }
 
     public static Throwable from(Throwable throwable) {
         String message = String.format("%s [%s] occurred while processing thread (%s)",
                 highLevelType(throwable), throwable, Thread.currentThread().getName());
-        return new SuppressedException(message, throwable);
+        return new SuppressedException(message);
```
Comment (nziebart) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/util/SuppressedException.java]: is this actually what we want? it means we will lose stacktrace information for the suppressed exception, right?

Comment (fsamuel-bs) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/util/SuppressedException.java]: Solution: add stacktrace to exception's message! Not the cleanest, but would actually work. What do you think?

Comment (schlosna) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/util/SuppressedException.java]: I don't think we need to add the stacktrace to the message. The intent behind this poorly named class was to propagate additional info via suppressed exceptions (I'll take the git blame). It should probably just be a static private class.

File: atlasdb-client/src/main/java/com/palantir/atlasdb/util/SuppressedException.java [hunk 2]
```diff
@@ -15,18 +15,21 @@
  */
 package com.palantir.atlasdb.util;
 
+import org.apache.commons.lang.exception.ExceptionUtils;
```
Comment (gsheasby) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/util/SuppressedException.java]: We should use `lang3` instead of `lang` here.

**Pull request merged**